FantasmoSDK/app/src/main/java/com/fantasmo/sdk/models/tensorflowML/ImageQualityModelUpdater.kt

```diff
@@ -90,16 +74,30 @@ class ImageQualityModelUpdater(val context: Context) {
 
     /**
      * Method that delivers an `Interpreter` with the model loaded onto it.
-     * First checks if the global interpreter has been loaded into memory.
-     * In negative case, it will check for updates and load those updates
-     * into memory.
+     * Checks if there's a config change. If true, it will request a new
+     * model. Else if there's no config change it will try to load a interpreter
+     * that has been loaded into memory. In negative case, it will check if the
+     * updates are loaded and use those instead.
      * @return `Interpreter` with model loaded
      */
     fun getInterpreter(): Interpreter? {
-        return if (::interpreter.isInitialized) {
-            interpreter
+        val remoteConfig = RemoteConfig.remoteConfig
+        if (remoteConfig.imageQualityFilterModelUri != null &&
+            remoteConfig.imageQualityFilterModelVersion != null &&
+            remoteConfig.imageQualityFilterModelVersion != modelVersion
```

Comment:
nice, so this means decreasing the version number will also trigger another download - that's good 👍🏾